grabbit/src/main/groovy/com/twcable/grabbit/jcr/JCRNodeDecorator.groovy

```diff
@@ -35,12 +38,27 @@ class JCRNodeDecorator {
     JCRNode innerNode
 
 
+    //Evaluated in a lazy fashion
+    private Collection<JCRNodeDecorator> immediateChildNodes
```

Comment:
For performance, I'd like immediate children to be evaluated in a lazy fashion (as below).